packages/runtime-core/src/components/Teleport.ts

```diff
@@ -119,6 +119,9 @@ export const TeleportImpl = {
         // Teleport *always* has Array children. This is enforced in both the
         // compiler and vnode children normalization.
         if (shapeFlag & ShapeFlags.ARRAY_CHILDREN) {
+          if (parentComponent && parentComponent.isCE) {
```

Comment:
Does it need to consider the same logic during hydrate teleport?

Reply:
sorry, I'm not very familiar with "hydration," but I've added a new E2E test case. I'm not sure if this test case can verify whether hydration is successful.

Reply:
That doesn't seem right, you didn't even assign a value to `teleportTarget` here
https://github.com/vuejs/core/blob/928af5fe2f5f366b5c28b8549c3728735c8d8318/packages/runtime-core/src/components/Teleport.ts#L435

Reply:
Great work !